doc/specs/Proto extensions-spec.md

```diff
@@ -0,0 +1,139 @@
+to add json snippets to our json files, and will be used when we generate settings for our various profiles. 
+
+## Inspiration
+
+### Goal: Allow programs to have a say in the settings for their profiles
+
+Currently, Ubuntu/WSL/Powershell etc are unable to provide any specifications for how they want
+their profiles in Terminal to look - only we and the users can modify the profiles. We want to provide
+these installations with the functionality to have a say in this, allowing them to specify things like
+their icon, their font and so on. However, we want to maintain that the user has final say over all of
+these settings. 
+
+## Solution Design
+
+Currently, when we load the settings we perform the following steps (this is a simplified description,
+for the full version see the spec for cascading default + user settings):
+
+1. Generate profiles from the defaults file
+2. Generate profiles from the dynamic profile generators
+3. Layer the user settings on top of all the profiles created in steps 1 and 2
+4. Validate the settings
+
+To allow for installations to add in their snippets of json, I propose the addition of a new step
+in between 2 and 3:
+
+1. Generate profiles from the defaults file
+2. Generate profiles from the dynamic profile generators
+3. **Incorporate the additional provided json stubs** - these stubs could be:
+   1. modifications to existing profiles
+   2. full profiles
+   3. colour schemes/modifications to colour schemes
+4. Layer the user settings on top of all the profiles created in steps 1 through 3
+5. Validate the settings
+
+### Specifications of the json stubs
+
+As written above, the json stubs could be modifications to existing profiles, full profiles, or colour schemes.
+
+#### Modifications to existing profiles
+
+For modifications to existing profiles, the json stub would need to indicate which profile it wishes to modify.
+A straightforward way to do this would be by providing the name of the profile. Since this modification would
+occur at the step _before_ any user changes to names are layered, the names of the profiles are consistent and
+can be used for identification. For example, the name of a profile generated by the WSL profile generator is
+simply the argument to "wsl.exe -d \<name\>", the name of a profile generated by the Powershell profile generator
+would be "Powershell Core" or "Powershell (ARM)" etc depending on the version.
+
+We might run into the case where multiple json stubs modify the same profile and so they override each other. We
+will need some sort of hierarchy to determine which stub to ultimately use. My initial thought is that we could
+check the owner of the json file and see if it matches the profile they are trying to modify (so something like
+only Powershell can modify the Powershell profiles and only Ubuntu can modify the Ubuntu profile etc). Independent
+parties will need to create full profiles with the desired commandline argument instead.
+
+#### Full profile stubs

```

Comment:
After a user installs a profile from a stub, the user's profiles.defaults get applied. Is there a way for the user to apply more customizations onto the newly created profile?

Reply:
Yes

> As in the case of the dynamic profile generator, if we create a profile that did not exist before (i.e. does not
exist in the user settings), we need to add the profile to the user settings file and re-save that file. 

So the user can just add more customizations as normal